scripts/runtestsuite/runtestsuite.py

```diff
@@ -237,6 +275,16 @@ def report_detailed_failure(self, scene, reference_filepath, output_filepath, lo
 
     def __write_header(self, args):
         script_path = os.path.realpath(__file__)
+        git_hash = git_title = "N/A"
+
+        if git_installed(self.template_directory):
+            if under_git_control(self.template_directory):
+                git_hash = get_git_hash(self.template_directory)
+                git_title = get_git_title(self.template_directory)
+            else:
+                git_hash = git_title = "N/A (Directory not under git control)"
+        else:
+            git_hash = git_title = "N/A (Git not installed)"
```

Comment:
Do you think this is easier to understand ? 

```suggestion

        is_git_installed = git_installed(self.template_directory)
        is_git_repository = under_git_control(self.template_directory)
        git_hash = git_title = None
        
        if is_git_installed and is_git_repository:      
            git_hash = get_git_hash(self.template_directory)
            git_title = get_git_title(self.template_directory)
        elif not is_git_installed:
            git_hash = git_title = "N/A (Git not installed)"
        else:
            git_hash = git_title = "N/A (Directory not under git control)"            
```